lib/src/test/java/com/thenewboston/api/bankapi/datasource/BankDataSourceTest.kt

```diff
@@ -2,59 +2,67 @@ package com.thenewboston.api.bankapi.datasource
 
 import com.thenewboston.common.http.NetworkClient
 import com.thenewboston.common.http.Outcome
-import com.thenewboston.common.http.config.BankConfig
 import com.thenewboston.common.http.config.Config
+import com.thenewboston.utils.BankApiMockEngine
 import io.kotest.matchers.collections.shouldNotBeEmpty
 import io.kotest.matchers.longs.shouldBeGreaterThan
 import io.kotest.matchers.should
 import io.kotest.matchers.string.contain
 import io.kotest.matchers.types.beInstanceOf
 import io.ktor.util.*
 import io.ktor.utils.io.errors.*
-import kotlinx.coroutines.runBlocking
-import org.junit.jupiter.api.DisplayName
-import org.junit.jupiter.api.Nested
-import org.junit.jupiter.api.Test
-import org.junit.jupiter.api.TestInstance
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.jupiter.api.*
 
 @KtorExperimentalAPI
 class BankDataSourceTest {
 
-    private var networkClient = NetworkClient(
-        BankConfig(
-            ipAddress = Config.IP_ADDRESS,
-            port = Config.PORT,
-            protocol = Config.PROTOCOL
-        )
-    )
+    @MockK
+    lateinit var networkClient: NetworkClient
 
-    private var bankDataSource = BankDataSource(networkClient)
+    private val mockEngine = BankApiMockEngine()
+
+    private lateinit var bankDataSource: BankDataSource
+
+    @BeforeEach
+    fun setup() {
+        MockKAnnotations.init(this)
+
+        bankDataSource = BankDataSource(networkClient)
+    }
 
     @Nested
     @DisplayName("Given a valid request that should return success")
     @TestInstance(TestInstance.Lifecycle.PER_CLASS)
     inner class GivenValidRequests {
 
         @Test
-        fun `test fetch list of available banks`() = runBlocking {
+        fun `test fetch list of available banks`() = runBlockingTest {
+            every { networkClient.defaultClient } returns mockEngine.getSuccess()
```

Comment:
This can be in a @BeforeAll within the GivenValidRequest once because it's the setup for all test cases in this nested test class.